src/test/java/net/minecraftforge/debug/world/SandboxChunkGeneratorTest.java

```diff
@@ -0,0 +1,162 @@
+import net.minecraft.world.level.biome.BiomeManager;
+import net.minecraft.world.level.biome.Biomes;
+import net.minecraft.world.level.biome.FixedBiomeSource;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.chunk.ChunkAccess;
+import net.minecraft.world.level.chunk.ChunkGenerator;
+import net.minecraft.world.level.levelgen.GenerationStep;
+import net.minecraft.world.level.levelgen.Heightmap;
+import net.minecraft.world.level.levelgen.RandomState;
+import net.minecraft.world.level.levelgen.blending.Blender;
+import net.minecraft.world.level.levelgen.structure.StructureSet;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.datafix.event.DataFixEvent;
+import net.minecraftforge.eventbus.api.GenericEvent;
+import net.minecraftforge.eventbus.api.IEventBus;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
+import net.minecraftforge.registries.DeferredRegister;
+import net.minecraftforge.registries.RegistryObject;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.function.Consumer;
+
+@Mod(SandboxChunkGeneratorTest.MODID)
+public class SandboxChunkGeneratorTest
+{
+    public static final String MODID = "sandbox";
+
+    private static final DeferredRegister<Codec<? extends ChunkGenerator>> CHUNK_GENERATORS = DeferredRegister.create(Registry.CHUNK_GENERATOR_REGISTRY, MODID);
+    public static final RegistryObject<Codec<? extends ChunkGenerator>> TEST_CHUNK_GENERATOR = CHUNK_GENERATORS.register("test", TestChunkGenerator::makeCodec);
+
+    public SandboxChunkGeneratorTest() // invoked by forge due to @Mod
+    {
+        IEventBus modBus = FMLJavaModLoadingContext.get().getModEventBus();
+        IEventBus forgeBus = MinecraftForge.EVENT_BUS;
+
+        CHUNK_GENERATORS.register(modBus);
+        modBus.register(new DFUConfigurator());
+    }
+
+    public static class DFUConfigurator {
+
+        @SubscribeEvent
+        public final void configueDFUSchema(final DataFixEvent.ConfigureTaggedChoice<String> event) {
+            if (event.getSchemaClass() == V2832.class && Objects.equals(event.getName(), "type")) {
+                event.register("sandbox:test", DSL::remainder);
+            }
+        }
+    }
+
+    public static class TestChunkGenerator extends ChunkGenerator
+    {
+        public static Codec<? extends ChunkGenerator> makeCodec()
+        {
+            Codec<TestChunkGenerator> codec = RecordCodecBuilder.create(builder -> builder.group(
+                                                                                                  RegistryOps.retrieveRegistry(Registry.STRUCTURE_SET_REGISTRY).forGetter(generator -> generator.structureSets),
```

Comment:
Is there more code on this line? Oh, there is, it's just indented far enough it was off my screen.

It's test mod code so it's whatever but why no static single codec instance, like every other chunk generator?